### PR TITLE
NEOS-752: support for email transformer exclusion list

### PR DESCRIPTION
**Pull request by evisdrenova**

### CODE REVIEW

File: worker/internal/benthos/transformers/transform_email.go [hunk 1]
```diff
@@ -44,9 +45,29 @@ func init() {
...
 		}
 
+		eL, err := args.Get("exclusion_list")
+		if err != nil {
+			return nil, err
+		}
+
+		excl, ok := eL.([]any)
+		if !ok {
+			return nil, fmt.Errorf("error")
```
Comment (nickzelei) [inline worker/internal/benthos/transformers/transform_email.go:55]: just so this doesn't get lost: this error should be updated. 
Should do something like this:

```
fmt.Errorf("Expected []any, got: %T", excl)
```

File: worker/internal/benthos/transformers/transform_email.go [hunk 2]
```diff
@@ -44,9 +45,29 @@ func init() {
...
+		}
+
+		var new []string
+
+		for _, a := range excl {
+			new = append(new, a.(string))
```
Comment (nickzelei) [inline worker/internal/benthos/transformers/transform_email.go:61]: need to check the type here.
```
strval, ok := a.(string)
if !ok {
  return nil, fmt.Errorf("Expected string, got :%T", a)
}
```

**Review by nickzelei — COMMENTED** [review]
left a few comments we talked about offline. overall would be great to have some better variable names for readability too

File: worker/internal/benthos/transformers/transform_email.go (hunk 1 above)
Comment (evisdrenova) [reply worker/internal/benthos/transformers/transform_email.go:55]: updated

File: worker/internal/benthos/transformers/transform_email.go (hunk 2 above)
Comment (evisdrenova) [reply worker/internal/benthos/transformers/transform_email.go:61]: udpated

**Pull request merged**